python/ray/tests/test_dashboard_profiler.py

```diff
@@ -92,6 +92,81 @@ def get_actor_flamegraph():
     )
 
 
+@pytest.mark.skipif(
+    os.environ.get("RAY_MINIMAL") == "1",
+    reason="This test is not supposed to work for minimal installation.",
+)
+@pytest.mark.skipif(sys.platform == "win32", reason="No memray on Windows.")
+@pytest.mark.parametrize("leaks", ["0", "1"])
+def test_memory_profiler_endpoint(ray_start_with_dashboard, leaks):
```

Comment:
feel like number of tests are not enough. 

1. we should test the failure cases (ideally by unit testing MemoryProfileManager)
2. Test cases incorrect inputs are passed (i.e., node ip)
3. Test cases where attach is requested while the previous run is not finished

Reply:
moved 1,3 to `test_profile_manager` unit test and 2 to `test_reporter`